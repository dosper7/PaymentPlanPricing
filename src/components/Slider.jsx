import React from 'react';

const Slider = (props) => {

    const OnValueChange = (evt) => {
        props.OnSliderChange(parseInt(evt.target.value));
    }

    return (
        
            <div>
                <div className="sliderRangeLabels text-primary">
                    <span>{props.Min}</span>
                    <span>{props.Max}</span>
                </div>
                <div className="slider">
                    <input type="range" min={props.Min} max={props.Max} step={props.Step} onChange={OnValueChange} value={props.SelectedLoanAmount}></input>
                </div>
                {props.children}
            </div>


    );
}

export default Slider;