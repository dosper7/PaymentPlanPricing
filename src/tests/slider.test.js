import React from 'react';
import Slider from '../components/Slider.jsx';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe(' <Slider/> testing', () =>{


    it('should render a slider',() =>{
        const slider = shallow(<Slider SelectedLoanAmount={1000} Min={100} Max={2000} Step={100}/>);
        expect(slider).toMatchSnapshot();
    });

    it('should contain a input type range',() =>{
        const slider = shallow(<Slider/>);
        expect(slider.find('input[type="range"]').length).toEqual(1);
    });

    it('input range should have the same value as passed in the prop "SelectedLoanAmount=1000" ',() =>{
        const slider = shallow(<Slider SelectedLoanAmount={1000}/>);
        slider.find('input[type="range"]').get(0).value=="1000";
    });
});


