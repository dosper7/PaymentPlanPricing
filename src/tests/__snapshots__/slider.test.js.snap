// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <Slider/> testing should render a slider 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slider
    Max={2000}
    Min={100}
    SelectedLoanAmount={1000}
    Step={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="sliderRangeLabels text-primary"
        >
          <span>
            100
          </span>
          <span>
            2000
          </span>
        </div>,
        <div
          className="slider"
        >
          <input
            max={2000}
            min={100}
            onChange={[Function]}
            step={100}
            type="range"
            value={1000}
          />
        </div>,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              100
            </span>,
            <span>
              2000
            </span>,
          ],
          "className": "sliderRangeLabels text-primary",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 100,
            },
            "ref": null,
            "rendered": 100,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2000,
            },
            "ref": null,
            "rendered": 2000,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            max={2000}
            min={100}
            onChange={[Function]}
            step={100}
            type="range"
            value={1000}
          />,
          "className": "slider",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "max": 2000,
            "min": 100,
            "onChange": [Function],
            "step": 100,
            "type": "range",
            "value": 1000,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sliderRangeLabels text-primary"
          >
            <span>
              100
            </span>
            <span>
              2000
            </span>
          </div>,
          <div
            className="slider"
          >
            <input
              max={2000}
              min={100}
              onChange={[Function]}
              step={100}
              type="range"
              value={1000}
            />
          </div>,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                100
              </span>,
              <span>
                2000
              </span>,
            ],
            "className": "sliderRangeLabels text-primary",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 100,
              },
              "ref": null,
              "rendered": 100,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2000,
              },
              "ref": null,
              "rendered": 2000,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              max={2000}
              min={100}
              onChange={[Function]}
              step={100}
              type="range"
              value={1000}
            />,
            "className": "slider",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "max": 2000,
              "min": 100,
              "onChange": [Function],
              "step": 100,
              "type": "range",
              "value": 1000,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
